### PR TITLE
bug fixed for http-server and ws-server

### PR DESCRIPTION
**Pull request by inhere**
http server:

- fix cannot setting params
- fix sometimes 'notFound' would be considered 'notAllowed'

ws server: 

- fix the connection info lost
- fix cannot connect to root path

devtool:

- fix devtool config reference is wrong

issues:

- fix https://github.com/swoft-cloud/swoft/issues/228

### CODE REVIEW

File: src/websocket-server/src/Router/Dispatcher.php [hunk 1]
```diff
@@ -31,15 +33,15 @@ public function handshake(Request $request, Response $response): array
...
+            list($className,) = $this->getHandler($path);
         } catch (\Throwable $e) {
             /* @var ErrorHandler $errorHandler */
             // $errorHandler = \bean(ErrorHandler::class);
             // $response = $errorHandler->handle($e);
             if ($e instanceof WsRouteException) {
                 return [
                     HandlerInterface::HANDSHAKE_FAIL,
-                    $response->withStatus(404)
+                    $response->withStatus(404)->withAddedHeader('Failed-Reason', 'Route not found')
```
Comment (huangzhhui) [inline src/websocket-server/src/Router/Dispatcher.php]: It should be `Failure-Reason`

File: src/websocket-server/src/Router/Dispatcher.php [hunk 2]
```diff
@@ -86,21 +88,34 @@ public function open(Server $server, Request $request, int $fd)
...
 
-        /** @var HandlerInterface $handler */
-        $handler = \bean($className);
-        $handler->onMessage($server, $frame);
+            /** @var HandlerInterface $handler */
+            $handler = \bean($className);
+            $handler->onMessage($server, $frame);
+        } catch (\Throwable $e) {
+            /** @var \Swoft\Event\EventManager $em */
+            $em = App::getBean('eventManager');
```
Comment (huangzhhui) [inline src/websocket-server/src/Router/Dispatcher.php]: 先判断 hasBean 再处理会好一点，因为是在 catch 里面，再有异常会重启 worker，可能会由于配置问题导致没有eventManager，特别是组件化拆分更细以后

File: src/websocket-server/src/WebSocketContext.php [hunk 3]
```diff
@@ -224,11 +259,12 @@ public static function getFdByCoId()
 
     /**
      * delete coId to fd mapping
+     * @param int|null $cid
      * @return bool
      */
-    public static function delFdToCoId(): bool
+    public static function delFdToCoId(int $cid = null): bool
```
Comment (huangzhhui) [inline src/websocket-server/src/WebSocketContext.php]: `deleteFdByCoId` 会更加的合适

File: src/websocket-server/src/WebSocketServer.php [hunk 4]
```diff
@@ -155,63 +172,67 @@ public function sendTo(int $receiver, string $data, int $sender = 0): int
 
         $this->log("(private)The #{$fromUser} send message to the user #{$receiver}. Data: {$data}");
 
-        return $this->server->push($receiver, $data, $opcode, $finish) ? 0 : -500;
+        return $this->server->push($receiver, $data, $opcode, $finish) ? 1 : 0;
     }
 
     /**
-     * broadcast message 广播消息
+     * broadcast message, will exclude self.
```
Comment (huangzhhui) [inline src/websocket-server/src/WebSocketServer.php]: will exclude itself ?

**Comment by inhere** [comment]
可以这几个点 我调整下

**Pull request merged**